source/common/http/http1/codec_impl.cc

```diff
@@ -295,8 +295,8 @@ void StreamEncoderImpl::endEncode() {
   flushOutput(true);
   connection_.onEncodeComplete();
   // With CONNECT or TCP tunneling, half-closing the connection is used to signal end stream.
-  if (connect_request_ || is_tcp_tunneling_) {
-    connection_.connection().close(Network::ConnectionCloseType::FlushWriteAndDelay);
+  if (connect_request_ || is_tcp_tunneling_ || is_response_to_connect_request_) {
+    connection_.connection().close(Network::ConnectionCloseType::FlushWrite);
```

Comment:
Can you add some comment trail here on why we are using flush write without delay and we used to do the delay? It's not immediately clear to me.